### PR TITLE
Add Rot64 custom gate to snarky

### PR DESCRIPTION
**Pull request by jspada**
Expose the the `Rot64` gate to snarky

Re https://github.com/MinaProtocol/mina/issues/12624

### CODE REVIEW

**Review by mrmr1993 — APPROVED** [review]

File: src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml [hunk 1]
```diff
@@ -188,7 +188,8 @@ module Plonk_constraint = struct
           ; v0c5 : 'v
           ; v0c6 : 'v
           ; v0c7 : 'v (* LSBs *)
-          ; compact : 'f
+          ; (* Coefficients *)
```
Comment (querolita) [inline src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml:191]: nice

File: src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml [hunk 2]
```diff
@@ -258,6 +259,38 @@ module Plonk_constraint = struct
           ; product1_lo : 'v
           ; product1_hi_0 : 'v
           }
+      | Rot64 of
+          { (* Current row *)
+            word : 'v
+          ; rotated : 'v
+          ; excess : 'v
+          ; bound_limb0 : 'v (* vpX are 12-bit plookup chunks *)
```
Comment (querolita) [inline src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml:267]: vpX?

**Review by querolita — APPROVED** [review]
Awesome! Looks great to me

File: src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml (hunk 2 above)
Comment (jspada) [reply src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml:267]: ups

Comment (jspada) [reply src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml]: Fixed!

**Comment by jspada** [comment]
!ci-build-me

**Comment by jspada** [comment]
!ci-build-me

**Pull request merged**